### PR TITLE
feat(clerk-react): Add 'nonce' prop to ClerkProvider

### PR DESCRIPTION
**Pull request by jescalan**
## Description

When using a [strict-dynamic](https://content-security-policy.com/strict-dynamic/) content security policy, Clerk currently will not work, as we do not pass a nonce to the hotloaded clerk-js script. This PR adds a `nonce` prop to `ClerkProvider` that can be used to thread the nonce value through to the clerk-js script load. It also includes an extra feature for next.js where the nonce will be automatically pulled from the CSP header and threaded through without needing any props so long as the provider is server-rendered.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/nextjs/src/app-router/server/utils.ts [hunk 1]
```diff
@@ -38,3 +38,48 @@ export const buildRequestLike = () => {
...
+  if (!nonce) {
+    return;
+  }
+
+  // Don't accept the nonce value if it contains HTML escape characters.
+  // Technically, the spec requires a base64'd value, but this is just an
+  // extra layer.
+  if (/[&><\u2028\u2029]/g.test(nonce)) {
+    throw new Error(
+      'Nonce value from Content-Security-Policy contained HTML escape characters.\nLearn more: https://nextjs.org/docs/messages/nonce-contained-invalid-characters',
```
Comment (BRKalow) [inline packages/nextjs/src/app-router/server/utils.ts:80]: Let's expand the error message and change the link here 👀

**Review by BRKalow — COMMENTED** [review]
Needs a changeset!

**Comment by nikosdouvlis** [comment]
Added a changeset to let the tests run :)

**Review by BRKalow — APPROVED** [review]
👏

**Pull request merged**